### PR TITLE
feat(deployment): ingest version checking

### PR DESCRIPTION
**Pull request by theosanderson**
We encountered an issue during a rollout in which the config map was changed and triggered a reload of the pod [edit: actually we didn't have the reloader - so I'm not sure if that is it] before the pod itself was upgraded to the next version, causing double version bumps in staging. This avoids that by making the config map contain the loculus version that it targets and making an initContainer in the deployment that checks that this matches that expected by the deployment.

https://ingest-version-checking.loculus.org/

### CODE REVIEW

**Comment by anna-parker** [comment]
Looks great! I see how this catches the case where we want to update the deployment before the config changes have rolled out. 
However I still wonder if it will catch the case where the config changes roll out before the deployment changes. This check only runs in the init step of creating a deployment. What happens when the config is updated using `reloader.stakater.com/auto`? The deployment will nto be affected, the config we mounted will still get updated, however we will not perform this check as the deployment will still be running. So we will be in the same situation as before... unless maybe I miss sth?

**Comment by theosanderson** [comment]
> What happens when the config is updated using reloader.stakater.com/auto? The deployment will nto be affected, the config we mounted will still get updated, however we will not perform this check as the deployment will still be running. So we will be in the same situation as before... unless maybe I miss sth?

So two things:

(A) this would work - what the reloader basically does is to trigger a new pod to be created - so the initContainers would run again
(B) I realise (just now) ingest isn't actually tagged with the reloader! We should probably fix that!

**Comment by anna-parker** [comment]
ah awesome! Didn't think that reloader caused initContainers to run again but you are correct!

File: kubernetes/loculus/templates/ingest-deployment.yaml [hunk 1]
```diff
@@ -24,6 +26,26 @@ spec:
         argocd.argoproj.io/sync-options: Force=true,Replace=true
```
Comment (anna-parker) [inline kubernetes/loculus/templates/ingest-deployment.yaml:26]: I think this where we need the annotation:
```suggestion
        argocd.argoproj.io/sync-options: Force=true,Replace=true
        reloader.stakater.com/auto: "true"
```

Comment (anna-parker) [reply kubernetes/loculus/templates/ingest-deployment.yaml:26]: Or no sorry this is a top level annotation so it should be in the metadata section not the metadata.template section. So maybe we should move both annotations up actually?

Comment (theosanderson) [reply kubernetes/loculus/templates/ingest-deployment.yaml]: Indeed!

**Review by anna-parker — APPROVED** [review]

**Pull request merged**